ingestion-beam/bin/download-schemas

```diff
@@ -1,12 +1,49 @@
 #!/bin/bash
 
-# Pulls down an archive of a test branch of mozilla-pipeline-schemas
-# for use in test suites.
+# Pulls down an archive of mozilla-pipeline-schemas and inserts test schemas for
+# use in test suites.
 
 set -e
 
+function init_working_directory() {
+    # Create a temporary directory for working. The current directory is stored
+    # into $rootdir and the working directory is stored into $workdir. The
+    # working directory will be removed on exit.
+    rootdir=$(pwd)
+    workdir=$(mktemp -d -t tmp.XXXXXXXXXX)
+    function cleanup {
+        rm -rf "$workdir"
+        echo "Running cleanup!"
+    }
+    trap cleanup EXIT
+    cd "$workdir" || exit
+}
+
+function insert_testing_schemas() {
+    # Patch the current schema archive with testing resources. This function
+    # should be called within the root directory of the extracted archive.
+    patch_url="https://github.com/jklukas/mozilla-pipeline-schemas/commit/9f8d4252a170ce9241a153b3b49eef235e1279a2.patch"
+    curl -sL ${patch_url} | git apply -v -p1
```

Comment:
This is some deep git that I would not have known to reach for. TIL.

For maintainability's sake, I think we may be better off avoiding patching in this commit and instead copy the contents directly into this script via heredocs.

Something like:

```
cat > schemas/test/test/test.1.schema.json << 'EOF'
{
  "$schema" : "http://json-schema.org/draft-04/schema#",
  "type" : "object",
  "title" : "default_schema",
  "properties" : {
  },
  "required" : []
}
EOF
```

I think that would let us get away from having to reference a commit from a random fork (which I again apologize for introducing as a test dependency here).

Reply:
The single quotes around EOF is a neat trick, I didn't know you could suppress parameter expansion that way.